### ISSUE TITLE
bazel build :all fails due to https://github.com/pubref/rules_protobuf 

### ISSUE THREAD

**Issue by AspirinSJL** [grpc/grpc#18256]
```
❯ bazel build :all                                                                             
ERROR: /usr/local/google/home/xxx/envs/grpc/grpc/worktree-h2/WORKSPACE:62:1: Traceback (most recent call last):
        File "/usr/local/google/home/xxx/envs/grpc/grpc/worktree-h2/WORKSPACE", line 62
                py_proto_repositories()
        File "/usr/local/google/home/xxx/.cache/bazel/_bazel_xxx/6ad28553f473ab366b04b5ff87b51d96/external/org_pubref_rules_protobuf/python/rules.bzl", line 13, in py_proto_repositories
                cpp_proto_repositories(**kwargs)
        File "/usr/local/google/home/xxx/.cache/bazel/_bazel_xxxl/6ad28553f473ab366b04b5ff87b51d96/external/org_pubref_rules_protobuf/cpp/rules.bzl", line 29, in cpp_proto_repositories
                fail(("Unknown loading rule %s for %s...)))
Unknown loading rule http_archive for {"url": "https://github.com/google/googletest/archive/7c6353d29a147cad1c904bf2957fd4ca2befe135.zip", "sha256": "f87029f647276734ef076785f76652347993b6d13ac1cbb2d2e976e16d2f8137", "strip_prefix": "googletest-7c6353d29a147cad1c904bf2957fd4ca2befe135", "name": "com_google_googletest"}
ERROR: Error evaluating WORKSPACE file
ERROR: Skipping ':all': while parsing ':all': error loading package 'external': Package 'external' contains errors
WARNING: Target pattern parsing failed.
ERROR: while parsing ':all': error loading package 'external': Package 'external' contains errors
INFO: Elapsed time: 0.042s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded)
```

Update from @jtattermusch: it seems that this is happening for bazel 0.23+ (build works fine for bazel 0.20, 0.21, 0.22 based on my testing)

**Comment by mhaidrygoog** [comment]
The issue here is that the WORKSPACE file under /usr/local which is specified in this error message is expecting only a grpc_archive rule while bazel is supplying a http_archive rule as well as a grpc_archive rule. The workaround here is to remove the else condition that fails the build if it doesn't receive a grpc_archive rule.
This way WORKSPACE file would ignore the http_archive rule and proceed with compilation on receiving the grpc_archive rule
Basically - 
 ```
 for dep in rem:
    rule = dep.pop("rule")
    if "grpc_archive" == rule:
      grpc_archive(**dep)
    else:
       fail("Unknown loading rule %s for %s" % (rule, dep))

```
To

 ```
 for dep in rem:
    rule = dep.pop("rule")
    if "grpc_archive" == rule:
      grpc_archive(**dep)
```

**Comment by jtattermusch** [comment]
I am experiencing the same issue.

**Comment by jtattermusch** [comment]
FYI @billfeng327

**Comment by jtattermusch** [comment]
A quick and dirty workaround to get C++ build working locally is to comment out this line https://github.com/grpc/grpc/blob/4982d915ccebd17492402b09e91435412ff9df29/WORKSPACE#L62

**Comment by jdwestbro** [comment]
I have a patch in the works that migrates the Bazel build to use https://github.com/stackb/rules_proto

**Comment by jdwestbro** [comment]
The fix for this is here:

https://github.com/jdwestbro/grpc/commit/f1ab0cdd8161266511dce0965c2bcb701e178167

and on Debian Rodete:

$ env CC=/usr/bin/gcc bazel build ...
INFO: Analysed 6673 targets (0 packages loaded, 0 targets configured).
INFO: Found 6673 targets...
INFO: Elapsed time: 11.366s, Critical Path: 0.00s
INFO: 0 processes.
INFO: Build completed successfully, 1 total action

PR pending...

**Comment by jtattermusch** [comment]
@mhaidrygoog  can come up with a PR that fixes this problem? It seems that you have some good insight into what's causing this problem.

@jdwestbro  is migrating over to stackb/rules_proto really necessary in order to fix the problem? (there might be good reasons to migrate over, but here we are mostly trying to fix the bazel build for bazel 0.23+)

**Comment by jtattermusch** [comment]
FTR pasting a link to the problematic rule https://github.com/pubref/rules_protobuf/blob/99043441e3c473809f633f5ad049945606124b60/cpp/rules.bzl#L27

**Comment by jtattermusch** [comment]
@gnossen  it seems that this problem is because of the ways our python bazel build is setup (see `py_proto_repositories()` in our WORKSPACE - it might need some massaging). Can you help?

**Comment by jdwestbro** [comment]
Sure if there's a simpler fix let's go with that. I thought stackb/rules_proto would be a drop-in replacement but there are some issues to resolve with protobuf code generation.

**Comment by gnossen** [comment]
Looking into it.
